### DISCUSSION TITLE
How to print the LaTeX text of the determinant

### DISCUSSION THREAD

**Discussion by HelloMrGeorge** [sympy/sympy#22031] [Q&A] [answered]
I found that in latex module 
`latex(Matrix([[1, 2], [3, 4]]), mat_delim='(')`
This statement can change delimiter from '[' to '(', but there is no delimiter like '|'.
What should I to print a determinant?

**Comment by jksuom** [comment, marked as answer]
'|' can be added to the delimiter dictionary:
```
--- a/sympy/printing/latex.py
+++ b/sympy/printing/latex.py
@@ -201,7 +201,7 @@ def __init__(self, settings=None):
                 self._settings['mul_symbol_latex_numbers'] = \
                     self._settings['mul_symbol']
 
-        self._delim_dict = {'(': ')', '[': ']'}
+        self._delim_dict = {'(': ')', '[': ']', '|': '|'}
 
         imaginary_unit_table = {
             None: r"i",
```

**Reply by HelloMrGeorge** [reply]
Thank you for your answer. I checked Sympy's source and found that there were only two keys in the dictionary as you said. Will Sympy try to add the delimiter of the determinant to the dictionary in the future? And now, if I'm going to print the determinant, does that mean I need to new an object and edit its dictionary?

**Reply by jksuom** [reply]
>  if I'm going to print the determinant, does that mean I need to new an object and edit its dictionary?

There is no need to create a new matrix. The single change in `latex.py` will make it possible to use `mat_delim='|'` in the `latex()` call with any existing matrix object.

**Reply by jksuom** [reply]
> Will Sympy try to add the delimiter of the determinant to the dictionary in the future? 

That should be done.

**Reply by oscarbenjamin** [reply]
Also `Determinant` should be made to print like that.

**Comment by oscarbenjamin** [comment]
There already is `Determinant` although it prints differently:
```python
In [1]: Determinant(Matrix([[1, 2], [3, 4]]))
Out[1]: 
│⎡1  2⎤│
│⎢    ⎥│
│⎣3  4⎦│
```

**Reply by asmeurer** [reply]
We should fix that. The determinant is never written like that.
